src/main/java/org/openrewrite/staticanalysis/RemoveToStringCallsFromArrayInstances.java

```diff
@@ -117,13 +127,8 @@ public J.Binary visitBinary(J.Binary binary, ExecutionContext ctx) {
             Expression left = binary.getLeft();
             Expression right = binary.getRight();
 
-            Cursor c = getCursor().dropParentWhile(is -> is instanceof J.Parentheses || !(is instanceof Tree));
-            if (c.getValue() instanceof J.MethodInvocation) {
-                if (PRINT_MATCHER.matches((J.MethodInvocation) c.getValue())) {
-                    if (left.getType() instanceof JavaType.Array || right.getType() instanceof JavaType.Array) {
-                        getCursor().putMessage("BINARY_FOUND", binary);
-                    }
-                }
+            if (binary.getOperator() == J.Binary.Type.Addition || left.getType() instanceof JavaType.Array || right.getType() instanceof JavaType.Array) {
```

Comment:
The binary must be using `+`:
```suggestion
            if (binary.getOperator() == J.Binary.Type.Addition && (left.getType() instanceof JavaType.Array || right.getType() instanceof JavaType.Array)) {
```